pkg/api/store/cluster/cluster_store.go

```diff
@@ -22,6 +26,101 @@ type Store struct {
 	mu           sync.Mutex
 }
 
+func SetClusterStore(ctx context.Context, schema *types.Schema, mgmt *config.ScaledContext) {
+	t := &transform.Store{
+		Store: schema.Store,
+		Transformer: func(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}, opt *types.QueryOptions) (map[string]interface{}, error) {
+			data, err := transformSetNilSnapshotFalse(data)
+			if err != nil {
+				return nil, err
+			}
+
+			return data, nil
+		},
+	}
+
+	s := &Store{
+		Store: t,
+	}
+
+	schema.Store = s
+}
+
+func transformSetNilSnapshotFalse(data map[string]interface{}) (map[string]interface{}, error) {
+	appliedVal, ok := data["appliedSpec"]
+	if ok {
+		appliedAsMap, ok := appliedVal.(map[string]interface{})
```

Comment:
there is a `convert` util library that can be used for to map conversion. Example:

https://github.com/rancher/rancher/blob/master/pkg/api/store/workload/workload_store.go#L180